c_plus_plus/src/model/stream/stream.cpp

```diff
@@ -85,14 +92,49 @@ void Stream::start()
 {
     mediaThread_->start();
     detectionThread_->start();
+    odasClient_->start();
+
+    status_ = StreamStatus::RUNNING;
+    emit statusChanged();
 }
 
 void Stream::stop()
 {
-    detectionThread_->stop();
-    detectionThread_->join();
-    mediaThread_->stop();
-    mediaThread_->join();
+    status_ = StreamStatus::STOPPING;
+    emit statusChanged();
+
+    if (odasClient_ != nullptr && odasClient_->getState() != OdasClientState::CRASHED)
+    {
+        odasClient_->stop();
+        odasClient_->join();
+    }
+
+    if (detectionThread_ != nullptr)
+    {
+        detectionThread_->stop();
+        detectionThread_->join();
+    }
+    if (mediaThread_ != nullptr)
+    {
+        mediaThread_->stop();
+        mediaThread_->join();
+    }
+    status_ = StreamStatus::STOPPED;
+    emit statusChanged();
 }
 
+void Stream::updateObserver()
+{
+    if (odasClient_ == nullptr) return;
+    const OdasClientState& state = odasClient_->getState();
```

Comment:
Why this is perfectly fine, I personally prefer not using the const ref in such case as it doesn't really provide benefit and makes the code (a bit) harder to read.